services/thermal_mgr/thermal_mgr.c

```diff
@@ -44,17 +45,60 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
 
+  if (thermalMgrQueueHandle == NULL || event == NULL) {
+    return ERR_CODE_INVALID_STATE;
+  }
+
+  error_code_t sendErrorCode = xQueueSend(thermalMgrQueueHandle, event, (TickType_t) 0);
```

Comment:
The xQueueSend function doesn't return an error_code_t but a different type of error. The easier way to handle this is just to move this so that its inside the if statement comparison so you don't need to care about the type of the return value